db/query.go

```diff
@@ -186,7 +204,26 @@ var QueryRolesExcludeDeleted = SGQuery{
 		base.KeyspaceQueryAlias,
 		base.KeyspaceQueryToken, base.KeyspaceQueryAlias,
 		base.KeyspaceQueryAlias, SyncDocWildcard,
-		base.KeyspaceQueryAlias, `\\_sync:role:%`,
+		base.KeyspaceQueryAlias, SyncRoleWildcard,
+		base.KeyspaceQueryAlias, base.KeyspaceQueryAlias,
+		base.KeyspaceQueryAlias, QueryParamStartKey,
+		base.KeyspaceQueryAlias),
+	adhoc: false,
+}
+
+var QueryRolesExcludeDeletedUsingRoleIdx = SGQuery{
+	name: QueryTypeRolesExcludeDeleted,
+	statement: fmt.Sprintf(
+		"SELECT META(%s).id "+
+			"FROM %s AS %s "+
+			"USE INDEX($idx) "+
```

Comment:
It looks like we're defining a separate query here because QueryRolesExcludeDeleted includes the SyncDocWildcard check in the WHERE clause, is that accurate?  I'm curious why the existing QueryUsers does not have that in the WHERE clause - any idea?  If it's not actually necessary in the roles query, it would allow us to use the same query against either index.

Reply:
I will need to test it to see why QueryRoles need SyncDocWildCard in where clause.